nixos/tests/postrunner.nix

```diff
@@ -0,0 +1,38 @@
+# run this testcae via nix-build nixos/tests/postrunner.nix
+import ./make-test-python.nix ({ pkgs, ... }: {
+name = "postrunner";
+meta = with pkgs.lib.maintainers; { maintainers = [ ngiger ]; };
+
+nodes.machine = { pkgs, ... }: {
+  environment.systemPackages = with pkgs; [ postrunner ];
+  };
+
+testScript = ''
+    start_all()
+
+    with subtest("Create user and log in"):
+        machine.wait_for_unit("multi-user.target")
+        machine.wait_until_succeeds("pgrep -f 'agetty.*tty1'")
+        machine.succeed("useradd -m alice")
+        machine.succeed("(echo foobar; echo foobar) | passwd alice")
+        machine.wait_until_tty_matches(1, "login: ")
+        machine.send_chars("alice\n")
+        machine.wait_until_tty_matches(1, "login: alice")
+        machine.wait_until_succeeds("pgrep login")
+        machine.wait_until_tty_matches(1, "Password: ")
+        machine.send_chars("foobar\n")
+        machine.wait_until_succeeds("pgrep -u alice bash")
+        machine.screenshot("0_alice_logged_in")
+
+    with subtest("Use postrunner"):
+        machine.succeed("echo ${pkgs.postrunner.meta.version}")
+        machine.screenshot("1_should_be_postrunner_version")
+        machine.succeed("postrunner --version")
```

Comment:
A "test" like this can be implemented as an [installCheck](https://nixos.org/manual/nixpkgs/stable/#ssec-installCheck-phase).

Reply:
Sorry, but this is beyond my novice NixOS level.

As I understand, I should

1. Remove the file `nixos/tests/postrunner.nix ` and
2. Add somehow a derivation of bundlerApp where I add some lines like
```ruby
    doInstallCheck = true;
    installCheckPhase = ''
      if [[ "$out/bin/postrunner --version)" == "postrunner version 1.0.4" ]]; then
        echo 'postrunner smoke check passed'
      else
        echo 'postrunner smoke check failed'
        return 1
      fi
    '';
```

But I found no file under nixpks where a bundlerApp has a checkInstall. And I am not versed enough to make this happen.

Can you give me hint or a sketch of the solution?

Thanks in advance.

Reply:
Alternately, you can use `passthru.tests` with `runCommand` helper, see https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests